dbms/src/Common/SimpleFixedThreadPool.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/Exception.h>
+#include <Common/Logger.h>
+#include <Common/SimpleFixedThreadPool.h>
+#include <common/logger_useful.h>
+
+namespace DB
+{
+SimpleFixedThreadPool::SimpleFixedThreadPool(const std::string & name_, size_t m_size_)
+    : name(name_)
+    , m_size(m_size_)
+    , thread_mgr(newThreadManager())
+{
+    RUNTIME_ASSERT(m_size > 0);
+}
+
+void SimpleFixedThreadPool::schedule(ThreadManager::Job job)
+{
+    {
+        std::unique_lock lock(mutex);
+        has_free_thread.wait(lock, [this] { return active_jobs < m_size; });
+        ++active_jobs;
+    }
+    thread_mgr->schedule(false, name, [&, job]() {
```

Comment:
https://github.com/pingcap/tiflash/blob/f49779562c9ad3233e69ebca526ed98fc500f57a/dbms/src/Common/ThreadManager.cpp#L46-L70
The memory of `futures` vector will keep growing forever.

Reply:
ok, updated.